job_list_params.go

```diff
@@ -0,0 +1,271 @@
+package river
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/riverqueue/river/internal/dbadapter"
+	"github.com/riverqueue/river/rivertype"
+)
+
+// JobListPaginationCursor is used to specify a starting point for a paginated
+// job list query.
+type JobListPaginationCursor struct {
+	id    int64
+	kind  string
+	queue string
+	time  time.Time
+}
+
+// JobListPaginationCursorFromJob creates a JobListPaginationCursor from a JobRow.
+func JobListPaginationCursorFromJob(job *rivertype.JobRow) *JobListPaginationCursor {
+	return &JobListPaginationCursor{
+		id:    job.ID,
+		kind:  job.Kind,
+		queue: job.Queue,
+		time:  jobListTimeValue(job),
+	}
+}
+
+// UnmarshalText implements encoding.TextUnmarshaler to decode the cursor from
+// a previously marshaled string.
+func (c *JobListPaginationCursor) UnmarshalText(text []byte) error {
```

Comment:
Just want to make sure I'm following: what's the purpose of this function? Is the idea here that it allows the cursor to be saved to JSON or otherwise persisted? Is that expected to be a feature that people want?

Reply:
`MarshalText` is the fallback for when a type does not implement `MarshalJSON`. This allows it to be marshaled and unmarshaled cleanly as an opaque JSON string, one which River can then ingest to set the cursor for the next page query.

In an API that powers a UI, for example, I'd expose the cursor so that the frontend can send it back with a subsequent request to fetch the next page.